src/SqlConnectionConfig.ts

```diff
@@ -67,12 +68,148 @@ export default class SqlConnectionConfig {
             throw new Error(`Invalid connection string. Please ensure 'Database' or 'Initial Catalog' is provided in the connection string.`);
         }
 
-        if (!this._connectionConfig.user) {
-            throw new Error(`Invalid connection string. Please ensure 'User' or 'User ID' is provided in the connection string.`);
+        const auth = this._connectionConfig['authentication'];
+        switch (auth?.type) {
+            case undefined: {
+                // SQL password
+                if (!this._connectionConfig.user) {
+                    throw new Error(`Invalid connection string. Please ensure 'User' or 'User ID' is provided in the connection string.`);
```

Comment:
should these errors be localized?

Reply:
Right now nothing is localized, and we haven't really had users requesting this either.